relays/client-substrate/src/client/caching.rs

```diff
@@ -161,6 +274,16 @@ impl<C: Chain, B: Client<C>> Client<C> for CachingClient<C, B> {
 		// since we have new underlying client, we need to restart subscriptions too
 		*self.data.grandpa_justifications.lock().await = None;
 		*self.data.beefy_justifications.lock().await = None;
+		// also restart background task too
+		*self.data.best_header.write().await = None;
+		*self.data.best_finalized_header.write().await = None;
```

Comment:
Normally not, but e.g. in case of chain restart it'll clear the old (invalidated) data. Without that, the relay will keep getting errors (because new headers will be from the new chain and old headers from the old chain version). Not that it is the super-important thing, but I'd prefer to be careful here :)